pkgs/development/python-modules/openmm/default.nix

```diff
@@ -0,0 +1,87 @@
+{ buildPythonPackage, lib, fetchFromGitHub, cmake, clang, gfortran, fftwSinglePrec
+, doxygen, ocl-icd, opencl-headers, swig, python, cython, numpy
+, cudatoolkit, enableCuda ? false
+} :
+
+buildPythonPackage rec {
+  pname = "openmm";
+  version = "7.7.0";
+
+  src = fetchFromGitHub {
+    owner = "openmm";
+    repo = pname;
+    rev = version;
+    hash = "sha256-2PYUGTMVQ5qVDeeABrwR45U3JIgo2xMXKlD6da7y3Dw=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    gfortran
+    swig
+    cython
+    doxygen
+  ];
+
+  buildInputs = [
+    fftwSinglePrec
+    ocl-icd
+    opencl-headers
+  ] ++ lib.optional enableCuda cudatoolkit
+  ;
+
+  propagatedBuildInputs = [
+    python
+    numpy
+  ];
+
+  dontUseSetuptoolsBuild = true;
+  dontUsePipInstall = true;
+  dontUseSetuptoolsCheck = true;
+
+  cmakeFlags = [
+    "-DCMAKE_C_COMPILER=${clang}/bin/clang"
+    "-DCMAKE_CXX_COMPILER=${clang}/bin/clang++"
```

Comment:
Is there are a reason to use clang?

Reply:
There is: http://docs.openmm.org/7.1.0/userguide/library.html#installing-on-linux-and-mac

Basically performance on CPU is much better with Clang and I would argue that for a package like OpenMM this is of importance. I will make Clang optional but enable by default. :slightly_smiling_face: